content/4_Gold/DP_Bitmasks.mdx

```diff
@@ -66,33 +66,51 @@ where $S  \setminus \{i\}$ is the subset $S$ without city $i$.
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
-#define ll long long
 
-const int MX = (1 << 20);
-ll dp[MX][21];
-vector<int> adj[21];
-const ll MOD = (ll)1e9+7;
+using ll = long long;
+
+const int MAX_N = 20;
+const ll MOD = (ll) 1e9 + 7;
+
+ll dp[1 << MAX_N][MAX_N + 1];
+// adj[i] contains the cities that can fly to i
+vector<int> adj[MAX_N + 1];
 
 int main() {
-	int N, M; cin >> N >> M;
-	for(int i = 0; i < M; i++){
-		int a, b; cin >> a >> b;
-		adj[b].push_back(a);
+	int city_num;
+	int flight_num;
+	cin >> city_num >> flight_num;
+	for (int f = 0; f < flight_num; f++) {
+		int start, end;
+		cin >> start >> end;
+		adj[end].push_back(start);
 	}
+	
+	// NOTE: the bitmask is 0-indexed, but the cities aren't
```

Comment:
you might as well make the cities 0-indexed too

Reply:
ok i just didn't want to change the code too much